Small miscellaneous fixes and changes

sciwhiz12 opened the pull request:
This PR contains a few small miscellaneous fixes and changes.

## Commit List

- **Fix staff sign-off code causing out-of-bounds error**

  The original code queried the first item in the users without checking if the list has any contents. This led to an IndexOutOfBoundsException when querying an empty list, and also means that if the staff member used another emote other than the first one in the list (which _may_ be possible depending on configuration), it wouldn't register the staff sign-off.

- **Add millis threshold to disconnect message**

  This filters out 'normal' disconnections, avoiding warning logspam while making note of notably long disconnections. Disconnections are always logged to DEBUG in any case.

- **Adjust message checker to ignore inaccessible channels**

  A channel is inaccessible if the bot user cannot view the channel or see the message history in the channel.

- **Store longs instead of TextChannels in old-channels helper**

  Storing the text channel means that the object will live on even if JDA discards the object, and will be a constant burden in the bot's memory. Using the snowflake ID of the channel is better memory-wise, and avoids storing JDA entities long-term.

---

- **Store messages awaiting sign-off in in-memory set**

  Without this, the bot would always log about the message awaiting sign-off every time it gets a reaction above the deletion limit.

sciwhiz12 committed:
Fix staff sign-off code causing out-of-bounds error

The original code queried the first item in the users without checking
if the list has any contents. This led to an IndexOutOfBoundsException
when querying an empty list, and also means that if the staff member used
another emote other than the first one in the list (which _may_ be
possible depending on configuration), it wouldn't register the staff
sign-off.

sciwhiz12 committed:
Add millis threshold to disconnect message

This filters out 'normal' disconnections, avoiding warning logspam while
making note of notably long disconnections. Disconnections are always
logged to DEBUG in any case.

sciwhiz12 committed:
Adjust message checker to ignore inaccessible channels

A channel is inaccessible if the bot user cannot view the channel or
see the message history in the channel.

sciwhiz12 committed:
Store longs instead of TextChannels in old-channels helper

Storing the text channel means that the object will live on even
if JDA discards the object, and will be a constant burden in the
bot's memory. Using the snowflake ID of the channel is better memory-wise,
and avoids storing JDA entities long-term.

KiriCattus reviewed (approved):
All looks good to me, do you have merge access or would you like me to merge it?

sciwhiz12 commented:
Holding off on merging (which I do have access for), as there's a fix which I want to sneak it, relating to the staff sign-off being a bit spammy.

sciwhiz12 committed:
Store messages awaiting sign-off in in-memory set

Without this, the bot would always log about the message awaiting sign-off
every time it gets a reaction above the deletion limit.

antoinegag commented:
Thank you for fixing my mess 😁
